rust/lance/src/index/vector/ivf/builder.rs

```diff
@@ -126,6 +132,76 @@ pub async fn shuffle_dataset(
         .await?)
 }
 
+pub async fn shuffle_dataset_v2(
```

Comment:
the `shuffle_dataset` is not used anymore, right? Lets just replace it?

Reply:
My two cents is to keep it, since the cleaner way of doing things is to move what I wrote into a datafusion node. Once we do that, we can go back to the original and delete v2.